ci: run legacy saucelabs for every build

devversion opened the pull request:
Recently we moved the Saucelabs job into a cronjob in order to avoid
heavy flakiness that we experienced due to a Saucelabs connect bug
that has been supposedly fixed by the Saucelabs team (no new version
is released yet though).

Our initial assumption was that we very rarely hit specific browser failures
and can therefore move the Saucelabs tests into a cronjob, but after some
days of having the cronjob, we realized that we actually hit browser-specific
failures quite often and that we should run the tests for every PR (like before)

Additionally renames the `test_saucelabs` job to contain `bazel`, as this is still
an experiment and we haven't fully replaced the legacy tests with the Bazel
saucelabs tests. Meanwhile we keep the Bazel tests in the cronjob as these
are just a subset of the actual legacy tests.

devversion commented:
@gkalpak I already hit a flake with the upgrade tests in this PR. Did you have some time to look into that test yet? Didn't we consider temporarily disabling this test if we don't get it stable?

gkalpak commented on .circleci/config.yml:
```diff
@@ -592,6 +592,9 @@ workflows:
       - test_aio:
           requires:
           - setup
+      - legacy-unit-tests-saucelabs:
```
Should this be a prerequisite for `publish_snapshot`?

devversion replied:
I think so. Let's add it.

gkalpak commented:
We are still waiting for a Zone.js release (https://github.com/angular/angular/pull/28219#discussion_r255574411), but it seems to be close now.
I don't think it is a specific test that fails - we would have to disable most ngUpgrade tests.

Alternatively, we could change the tests to only run on one version of AngularJS (or find a more robust way to run against multiple versions).

devversion committed:
ci: run legacy saucelabs for every build

Recently we moved the Saucelabs job into a cronjob in order to avoid
heavy flakiness that we experienced due to a Saucelabs connect bug
that has been supposedly fixed by the Saucelabs team (no new version
is released yet though).

Our initial assumption was that we very rarely hit specific browser failures
and can therefore move the Saucelabs tests into a cronjob, but after some
days of having the cronjob, we realized that we actually hit browser-specific
failures quite often and that we should run the tests for every PR (like before)

kara commented:
@devversion Same deal here. Needs a separate PR for patch branch :(
